src/components/calendar/calendar.tsx

```diff
@@ -6,45 +6,51 @@ import styles from './calendar.module.scss';
 
 interface CalendarComponentProps {
     selectedDate: Date;
-    setSelectedDate: (date: Date | undefined) => void;
-    availableDates: string[];
+    onDateSlected: (date: Date) => void;
+    availableDates: Date[];
+    setSelectedHour: (hour: string) => void;
```

Comment:
the calendar doesn't know what is an hour..
the lesson page should take care of that